src/pretix/base/views/metrics.py

```diff
@@ -0,0 +1,35 @@
+from django.conf import settings
+from http import HttpResponse
+from .. import metrics
+
+unauthedResponse = HttpResponse("<html><title>Forbidden</title><body>You are not authorized to view this page.</body></html>", mimetype="text/html")
+unauthedResponse["WWW-Authenticate"] = 'Basic realm="metrics"'
+unauthedResponse.status_code = 401
```

Comment:
I believe it is a bad Idea to have this ``HttpResponse`` as a long-living object as it is mutable and might be mutated e.g. by middlewars. I suggest to replace it with a ``reponse_unauthenticated()`` method that returns this object.